### PR TITLE
Remove model settings subsection

### PR DESCRIPTION
**Pull request by gassmoeller**
This is a rework of #2105, but without its dependencies on #2051 and #2099. It only depends on #2156, which should be easy to merge. It leaves all of the class structure intact, and only moves parameters in different subsections. The most complicated part is really the parameter update script.

### CODE REVIEW

File: source/simulator/parameters.cc [hunk 1]
```diff
@@ -550,45 +520,40 @@ namespace aspect
...
-                         "of the Stokes equation. This feature is likely only used when implementing force "
-                         "vectors for manufactured solution problems and requires filling additional outputs "
-                         "of type AdditionalMaterialOutputsStokesRHS.");
-
     }
-    prm.leave_subsection ();
+    prm.leave_subsection();
+
+
+    prm.declare_entry ("Remove nullspace", "",
```
Comment (naliboff) [inline source/simulator/parameters.cc]: Consider adding a separate subsection from the nullspace removal (`Nullspace removal`).

File: source/simulator/parameters.cc [hunk 2]
```diff
@@ -1503,7 +1470,7 @@ namespace aspect
     AssertThrow((!use_direct_stokes_solver) || (nullspace_removal == NullspaceRemoval::none),
                 ExcMessage("Because of the difference in system partitioning, nullspace removal is "
                            "currently not compatible with the direct solver. "
-                           "Please turn off one or both of the options 'Model settings/Remove nullspace', "
+                           "Please turn off one or both of the options 'Remove nullspace', "
```
Comment (naliboff) [inline source/simulator/parameters.cc]: Don't forgot to change this if you add the section `Nullspace removal`.

File: doc/update_prm_files_to_2.0.0.sed [hunk 3]
```diff
@@ -72,3 +76,214 @@ s/subsection Dynamic Topography/subsection Dynamic topography/g
 
 # Remove all instances of `melt transport threshold`
 /Melt transport threshold/d
+
+# This part deals with the old options in the 'Model settings' subsection.
+# They have been moved into more appropriate subsections, but sed is
+# not good at moving things above their current position. Instead
+# we store everything correctly formatted in the hold space and
+# add it to the end of the file. Since parameter files do not mind
```
Comment (naliboff) [inline doc/update_prm_files_to_2.0.0.sed]: Update

File: doc/update_prm_files_to_2.0.0.sed [hunk 4]
```diff
@@ -72,3 +76,214 @@ s/subsection Dynamic Topography/subsection Dynamic topography/g
...
+
+/^ *#/{
+# If there are comments add the next line
+N
+
+# If a comment was not followed by a 'set' we might have
+# picked up the 'end' that was supposed to stay there.
+# This is a problem, because we can not notice that the subsection
+# 'Model settings' is over if we do not find the 'end'.
+# Thus, replace the content of the pattern by a singe 'end' and
```
Comment (naliboff) [inline doc/update_prm_files_to_2.0.0.sed]: singe -> single.

File: doc/update_prm_files_to_2.0.0.sed [hunk 5]
```diff
@@ -72,3 +76,214 @@ s/subsection Dynamic Topography/subsection Dynamic topography/g
...
+end\
+/
+H
+d
+}
+
+/set Remove nullspace/ {
+s/^.*$/&\
+/
+s/ *set Remove nullspace/set Remove nullspace/
```
Comment (naliboff) [inline doc/update_prm_files_to_2.0.0.sed]: Update for new nullspace section.

File: doc/update_prm_files_to_2.0.0.sed [hunk 6]
```diff
@@ -72,3 +76,214 @@ s/subsection Dynamic Topography/subsection Dynamic topography/g
...
+
+
+
+# Now the Model settings subsection should be empty.
+# Remove any remaining lines (e.g. empty lines, comments),
+# except for the last one (to not jump over it).
+/^ *end/ !d
+
+/^ *end/ {
+# At the end of the file, add a comment and the content of the hold space
```
Comment (naliboff) [inline doc/update_prm_files_to_2.0.0.sed]: Update comment.

File: benchmarks/blankenbach/base_case1a.prm [hunk 7]
```diff
@@ -90,12 +90,15 @@ subsection Mesh refinement
   set Refinement fraction                      = 1.0
 end
 
-subsection Model settings
-  set Include adiabatic heating               = false
-  set Include shear heating                   = false
```
Comment (naliboff) [inline benchmarks/blankenbach/base_case1a.prm]: Make sure sed script fails if`set Include adiabatic heating` or `set Include shear heating` equal `true`.

File: cookbooks/convection-box.prm [hunk 8]
```diff
@@ -86,16 +86,15 @@ end
 # which boundaries actually carry a prescribed boundary temperature, whereas
 # all other parts of the boundary are insulated (i.e., no heat flux through
 # these boundaries; this is also often used to specify symmetry boundaries).
-subsection Model settings
+# The parameters below this comment were created by the update script
```
Comment (naliboff) [inline cookbooks/convection-box.prm]: Need to check that all comments in the cookbooks and benchmarks are preserved (fix in a later commit).

**Comment by jdannberg** [comment]
Can you also change the file that updates source files to include the changes to where the fixed boundary temperature/composition indicators are now located?

**Comment by naliboff** [comment]
Please merge yourself after check.sh has finished running.

**Comment by gassmoeller** [comment]
I ran `check.sh` in benchmarks and cookbooks, and all parameter files seem to work. This should be ready, as soon as #2162 and #2164 are merged (I will need to rebase afterwards).

**Pull request merged**